### DISCUSSION TITLE
make failed (Centos 7)

### DISCUSSION THREAD

**Discussion by jrtkcoder** [baresip/re#517] [General]
CC      build-x86_64/main/openssl.o
src/main/openssl.c: In function ‘openssl_init’:
src/main/openssl.c:30:2: warning: implicit declaration of function ‘OPENSSL_init_ssl’ [-Wimplicit-function-declaration]
  err = OPENSSL_init_ssl(OPENSSL_INIT_SSL_DEFAULT, NULL);
  ^
src/main/openssl.c:30:2: warning: nested extern declaration of ‘OPENSSL_init_ssl’ [-Wnested-externs]
src/main/openssl.c:30:25: error: ‘OPENSSL_INIT_SSL_DEFAULT’ undeclared (first use in this function)
  err = OPENSSL_init_ssl(OPENSSL_INIT_SSL_DEFAULT, NULL);
                         ^
src/main/openssl.c:30:25: note: each undeclared identifier is reported only once for each function it appears in
make: *** [build-x86_64/main/openssl.o] Error 1

**Comment by jrtkcoder** [comment]
openssl version:
OpenSSL 1.1.1c  28 May 2019

Linux version 3.10.0-1160.53.1.el7.x86_64 (mockbuild@kbuilder.bsys.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC) ) #1 SMP Fri Jan 14 13:59:45 UTC 2022

**Comment by sreimers** [comment]
Looks like the wrong OpenSSL headers are loaded. We dropped OpenSSL 1.0.x Support, I think this is the default on RedHat/Centos 7.

**Comment by jrtkcoder** [comment]
@sreimers I upgrade  the Openssl to 3.0.2, make failed again, show me the same error. 
![image](https://user-images.githubusercontent.com/19484957/188561154-ab3b9fd5-1bcc-4e35-aefe-1bda9d66ac55.png)

**Comment by sreimers** [comment]
`openssl` is only the binary, you need also the openssl dev/header files at the right place. 

You can give cmake a try:

```
cmake -B build
```

Or you can try to build OpenSSL static and give cmake a hint:

```
OPENSSL_ROOT_DIR=/path/to/openssl cmake -B build
cmake --build build -j
```

In general we dropped Centos 7 support some months ago. With a newer cmake, [compiler](https://github.com/baresip/re/wiki/Centos-7-Support) and a supported openssl version it can work, but nothing we test or support. [v2.3.0](https://github.com/baresip/re/releases/tag/v2.3.0) was the last supported release.

**Comment by robert-scheck** [comment]
@jrtkcoder, is there anything which prevents you from just using the [libre](https://packages.fedoraproject.org/pkgs/libre/libre/), [librem](https://packages.fedoraproject.org/pkgs/librem/librem/) and [baresip](https://packages.fedoraproject.org/pkgs/baresip/baresip/) RPM packages from EPEL 7? I've built the packages there, also using a more modern C compiler, against the [openssl11](https://packages.fedoraproject.org/pkgs/openssl11/openssl11/) RPM package (which provides the same OpenSSL version like in CentOS/RHEL/Rocky Linux 8).
